src/NUnitFramework/nunitlite/CommandLineOptions.cs

```diff
@@ -64,18 +66,102 @@ internal CommandLineOptions(IDefaultOptionsProvider defaultOptionsProvider, para
             
             ConfigureOptions();            
             if (args != null)
-                Parse(args);
+                Parse(PreParse(args));
         }
 
         public CommandLineOptions(params string[] args)
         {
             ConfigureOptions();
             if (args != null)
-                Parse(args);
+                Parse(PreParse(args));
         }
-        
+
+#if !PORTABLE
+        private int _nesting = 0;
+#endif
+
+        internal IEnumerable<string> PreParse(IEnumerable<string> args)
```

Comment:
This is another method that is very similar to the Expand function in nunit/nunit-console#98. @oznetmaster do you think you and @NikolayPianikov could work together to bring your two implementations closer together? If we find bugs in the future, it would be easier to fix it in multiple places if we were dealing with nearly identical code.